### ISSUE TITLE
Bug: Unescaping reserved character @

### ISSUE THREAD

**Issue by mrchief** [axios/axios#1212]
Summary
----
According to the [spec](https://tools.ietf.org/html/rfc3986#section-2.2), `@` is a reserved character and thus should be encoded. But axios un-encodes this: https://github.com/axios/axios/blob/master/lib/helpers/buildURL.js#L7

```js
function encode(val) {
  return encodeURIComponent(val).
    replace(/%40/gi, '@').
    replace(/%3A/gi, ':').
    replace(/%24/g, '$').
    replace(/%2C/gi, ',').
    replace(/%20/g, '+').
    replace(/%5B/gi, '[').
    replace(/%5D/gi, ']');
}
```
This seems like a bug. What's the reason?

Context
----
- axios version: *e.g.: v0.17.0*
- Environment: *e.g.: node v8.4.0, windows 7 x64

**Comment by arnav13081994** [comment]
I have the same problem.

**Comment by Jerome1337** [comment]
Same problem here, trying to execute request with `@` char and then Axios return ```Network error```

**Comment by jasonsaayman** [comment]
Solved by #1671
